mip/options.go

```diff
@@ -1,10 +1,203 @@
 package mip
 
-import "time"
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
 
-// Limits on the MIP solver, such as how long it can run for.
-type Limits struct {
+// SolveOptions are options that can be cionfigured for any solver.
+type SolveOptions struct {
 	// Duration is the maximum duration of the solver. A duration limit of 0 is
 	// treated as infinity.
-	Duration time.Duration `json:"duration" usage:"maximum duration of the solver. A duration limit of 0 is treated as infinity" default:"30s"`
+	Duration time.Duration `json:"duration" usage:"Maximum duration of the solver." default:"30s"`
+	// Verbosity of the solver in the console.
+	Verbosity Verbosity `json:"verbosity" usage:"{off, low, medium, high} Verbosity of the solver in the console." default:"off"`
+	// Mip-specific options.
+	Mip MipOptions `json:"mip" usage:"Options specific to MIP problems. Linear problems do not use these options."`
```

Comment:
To me it would make more sense if acronyms are all capital, so

` MIP MIPOptions`

As a side not, these MIPOptions are control options we expose as _hard coded_ options. I would not make a sub-category for the option bening a MIP option. I am pretty sure the users will know what a GAP is (if they touch this option).

Reply:
Changed it to MIP though.